# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1296242517 18000
#      Fri Jan 28 14:21:57 2011 -0500
# Branch release-3-4-x
# Node ID 81e9ff0cc729d2fb7ce9817227324aaa7fd8f263
# Parent  3f162dd0c112d1164b36ea312c580ca70f3e7fd2
eigs-base.cc: fix error in change removing HAVE_ARPACK

diff --git a/liboctave/ChangeLog b/liboctave/ChangeLog
--- a/liboctave/ChangeLog
+++ b/liboctave/ChangeLog
@@ -1,8 +1,13 @@
+2011-01-28  John W. Eaton  <jwe@octave.org>
+
+	* eigs-base.cc: Remove #endif corresponding to "#ifdef HAVE_ARPACK",
+	not the one that matched "#if !defined (CXX_NEW_FRIEND_TEMPLATE_DECL)".
+
 2011-01-28  John W. Eaton  <jwe@octave.org>
 
 	* eigs-base.cc (EigsRealNonSymmetricFunc,
 	EigsRealNonSymmetricMatrix, EigsRealNonSymmetricMatrixShift):
 	Initialize eig_vec2 to 0 before calling dneupd.
 
 2011-01-28  John W. Eaton  <jwe@octave.org>
 
diff --git a/liboctave/eigs-base.cc b/liboctave/eigs-base.cc
--- a/liboctave/eigs-base.cc
+++ b/liboctave/eigs-base.cc
@@ -196,16 +196,18 @@ static Matrix
 utsolve (const SparseMatrix&, const ColumnVector&, const Matrix&);
 
 static ComplexMatrix
 utsolve (const ComplexMatrix&, const ColumnVector&, const ComplexMatrix&);
 
 static Matrix
 utsolve (const Matrix&, const ColumnVector&, const Matrix&);
 
+#endif
+
 template <class M, class SM>
 static octave_idx_type
 lusolve (const SM& L, const SM& U, M& m)
 {
   octave_idx_type err = 0;
   double rcond;
   MatrixType utyp (MatrixType::Upper);
 
@@ -3822,17 +3824,16 @@ extern octave_idx_type
 EigsComplexNonSymmetricFunc (EigsComplexFunc fun, octave_idx_type n,
                              const std::string &_typ, Complex sigma,
                              octave_idx_type k, octave_idx_type p,
                              octave_idx_type &info, ComplexMatrix &eig_vec,
                              ComplexColumnVector &eig_val,
                              ComplexColumnVector &resid, std::ostream& os,
                              double tol = DBL_EPSILON, bool rvec = false,
                              bool cholB = 0, int disp = 0, int maxit = 300);
-#endif
 
 #ifndef _MSC_VER
 template static octave_idx_type
 lusolve (const SparseMatrix&, const SparseMatrix&, Matrix&);
 
 template static octave_idx_type
 lusolve (const SparseComplexMatrix&, const SparseComplexMatrix&,
          ComplexMatrix&);
